src/Stratis.Bitcoin.Features.Miner/Staking/PosMinting.cs

```diff
@@ -368,8 +368,8 @@ public async Task GenerateBlocksAsync(WalletSecret walletSecret, CancellationTok
                     continue;
                 }
 
-                // Don't stake when the wallet tip is not on the current chain.
-                if (this.consensusManager.Tip.FindAncestorOrSelf(this.walletManager.WalletTipHash) == null)
+                // Don't stake if the wallet is not up-to-date with the current chain.
```

Comment:
ahhh good spot so we just checked we where on the same chain? do we not have any check that we are near the tip? (like IBD or something?)

Reply:
There is an IBD check right after this check:

```                // Prevent staking if in initial block download.
                if (this.initialBlockDownloadState.IsInitialBlockDownload())
                {
                    this.logger.LogTrace("Waiting for synchronization before mining can be started.");

                    await Task.Delay(TimeSpan.FromMilliseconds(this.minerSleep), cancellationToken).ConfigureAwait(false);
                    continue;
                }
```